### ISSUE TITLE
Stuck at the step antsRegistation

### ISSUE THREAD

**Issue by XinyuLiu1999** [thalamicseg/hipsthomasdocker#1]
Hi, 

I am trying to run this tool for example data. However every time I ran it, I was stuck at this step:
![image](https://github.com/thalamicseg/hipsthomasdocker/assets/77971825/8565f6d1-4365-4dd3-b286-ae501faa82a7)
It takes forever and can not continue. I tried it with both MacOS and WSL and both didn't work.

Could you help resolving this?
Thank you very much!

Xinyu

**Comment by leRequinRomain** [comment]
I have the same issue. Did you ever find a solution?

![image](https://github.com/thalamicseg/hipsthomasdocker/assets/143903679/815b58ea-08e8-477f-9b65-50214e4452c9)

**Comment by XinyuLiu1999** [comment]
> I have the same issue. Did you ever find a solution?
> 
> ![image](https://user-images.githubusercontent.com/143903679/265224770-815b58ea-08e8-477f-9b65-50214e4452c9.png)

Hi,

After communication with the author this is due to insufficient RAM, since this step would take a significant amount of memory. One solution I found for my MacOS is to turn on the Rosetta option in docker, then it can be run smoothly within 1 hour. It is also recommended to run in a Linux computer rather than from a virtual machine etc, which will limit RAM usage.

**Comment by leRequinRomain** [comment]
After making the post, I let the program run for a long time. It eventually completed, but was stuck at this step for about 20-30 minutes. The same behavior occurred with both example scripts. I ran the program on Windows 10 Ubuntu WSL2 with 64 GB RAM.

**Comment by anagrammarian** [comment]
Yes this is the ANTS nonlinear registration step which can take anywhere from 10-30 min depending on your processor. It takes about 20 min on a cheap PC running Ubuntu and about 10 min on a ThinkStation which is more snazzy. There is an issue on Macs running Apple silicon and the fix is on the main thread (install rosetta and make sure it is enabled and make sure docker allocation of memory is increased, default is ridiculously small).

The file NULL does not exist is an ANTS warning which should not happen according to the ANTs developers but does as we are not passing real mask files (defaults to NULL). It is benign and will complete.
